scripts/ensure-venv.sh

```diff
@@ -2,40 +2,51 @@
 
 # optionally opt out of virtualenv creation
 # WARNING: this will be removed (most likely renamed) soon!
-if [ "$SENTRY_NO_VIRTUALENV_CREATION" == "1" ]; then
+if [[ "$SENTRY_NO_VIRTUALENV_CREATION" == "1" ]]; then
     exit 0
 fi
 
+yellow="$(tput setaf 3)"
+bold="$(tput bold)"
+reset="$(tput sgr0)"
+
 venv_name=".venv"
-python_bin="python2.7"
+python_version="3.6"
 
-if [ "$SENTRY_PYTHON3" = "1" ]; then
-    venv_name=".venv3"
-    python_bin="python3.6"
+if [[ "$SENTRY_PYTHON2" = "1" ]]; then
+    venv_name=".venv2"
+    python_version="2.7"
 fi
 
-die() {
-    cat >&2 "${@}"
+die () {
+    cat <<EOF
+$@
+EOF
     exit 1
 }
 
-if [ -n "$VIRTUAL_ENV" ]; then
-    # we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
-    if [ "$VIRTUAL_ENV" != "${PWD}/${venv_name}" ]; then
-        die "You're in a virtualenv, but it's not in the expected location (${PWD}/${venv_name})"
-    fi
-
-    # TODO: Update this to strictly check .python-version
-    if [ "$SENTRY_PYTHON3" = "1" ]; then
-        python -c "import sys; sys.exit(sys.version_info[:2] != (3, 6))" ||
-            die "For some reason, the virtualenv isn't Python 3.6."
+if [[ -n "$VIRTUAL_ENV" ]]; then
+    major=`python -c "import sys; print(sys.version_info[0])"`
+    minor=`python -c "import sys; print(sys.version_info[1])"`
+    # If .venv contains Python2 or SENTRY_PYTHON2 is set, then fail with instructions
+    if [[ "$major" -eq 2 ]] || [[ ! -z "$SENTRY_PYTHON2" ]]; then
+        cat << EOF
+${yellow}${bold}
+WARNING! You are running a Python 2 virtualenv, which is DEPRECATED.
+Support will be dropped beginning in 2021.
+${reset}
+EOF
     else
-        python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" ||
-            die "For some reason, the virtualenv isn't Python 2.7."
+        # If .venv is less than Python 3.6 fail
+        [[ "$minor" -lt 6 ]] &&
+            die "Remove $VIRTUAL_ENV and try again since the Python version installed should be at least 3.6."
     fi
 else
-    if [ ! -f "${venv_name}/bin/activate" ]; then
-        die "You don't seem to have a virtualenv. Please create one by running: ${python_bin} -m virtualenv ${venv_name}"
+    if [[ ! -f "${venv_name}/bin/activate" ]]; then
+        die "You don't seem to have a virtualenv. Please create one by running: python${python_version} -m virtualenv ${venv_name}"
     fi
     die "You have a virtualenv, but it doesn't seem to be activated. Please run: source ${venv_name}/bin/activate"
 fi
+
+# Somehow it does not succeed unless I exit with 0
```

Comment:
How did you arrive at this workaround? Under what circumstances was the script completing without `die` and exiting nonzero? I have a feeling it was some other factor, because this shouldn't be necessary.

Reply:
You will need to look at this output carefully (full output at the end):

First execution of `direnv allow` after commenting out the `exit 0` line.
```
direnv: Ensuring proper virtualenv...
direnv wasn't able to complete execution.
```

The second execution of `direnv allow` after adding back `exit 0` line.
```
direnv: Ensuring proper virtualenv...
direnv: Your virtualenv is activated, but sentry doesn't seem to be installed.
direnv: To install sentry, please type: make install-py-dev
```

```
❯ direnv allow
direnv: loading ~/code/sentry/.envrc
direnv: Configuring git...
direnv: Activating virtualenv...
direnv: You don't seem to have a virtualenv.
direnv: To create a virtualenv, please type: python3.6 -m virtualenv .venv
direnv wasn't able to complete execution.
You may have been given some recommendations in the error message.
Follow them, and then you'll need to redo direnv by running "direnv allow".

direnv tooling is in an ALPHA state!
If you're having trouble, or have questions, please ask in #discuss-dev-tooling
and/or reach out to @josh.

sentry on  dev/py3-default [$!] via ⬢ v12.19.0 via 🐍 v3.6.10
❯ python3.6 -m virtualenv .venv
created virtual environment CPython3.6.10.final.0-64 in 240ms
  creator CPython3Posix(dest=/Users/armenzg/code/sentry/.venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/armenzg/Library/Application Support/virtualenv)
    added seed packages: pip==20.2.4, setuptools==50.3.2, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

sentry on  dev/py3-default [$!] via ⬢ v12.19.0 via 🐍 v3.6.10
❯ direnv allow
direnv: loading ~/code/sentry/.envrc
direnv: Configuring git...
direnv: Activating virtualenv...
direnv: Ensuring proper virtualenv...
direnv wasn't able to complete execution.
You may have been given some recommendations in the error message.
Follow them, and then you'll need to redo direnv by running "direnv allow".

direnv tooling is in an ALPHA state!
If you're having trouble, or have questions, please ask in #discuss-dev-tooling
and/or reach out to @josh.

sentry on  dev/py3-default [$!] via ⬢ v12.19.0 via 🐍 v3.6.10
❯ direnv allow
direnv: loading ~/code/sentry/.envrc
direnv: Configuring git...
direnv: Activating virtualenv...
direnv: Ensuring proper virtualenv...
direnv: Your virtualenv is activated, but sentry doesn't seem to be installed.
direnv: To install sentry, please type: make install-py-dev
direnv wasn't able to complete execution.
You may have been given some recommendations in the error message.
Follow them, and then you'll need to redo direnv by running "direnv allow".
```

Reply:
In the interest of time I'll look at this later, this shouldn't block.